### PR TITLE
Splitted Aperture CR to Agent and Controller CR

### LINKED ISSUE #196: Split Agent and Controller from Aperture CRD
We need to separate the Agent and Controller from the current Aperture repo as there can be scenarios where user might want to install the agent in one cluster and controller in other.

### PR DESCRIPTION
**Pull request by hdkshingala**
Fixes #196

- - -
This change is [Reviewable](https://reviewable.io/reviews/fluxninja/aperture/206)

### CODE REVIEW

**Review by kklimonda-fn — APPROVED** [review]
> Reviewed 72 of 72 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/fluxninja/aperture/206)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @hdkshingala)

**Review by kwapik — CHANGES_REQUESTED** [review]
> Reviewed 72 of 72 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/fluxninja/aperture/206)* status: all files reviewed, 2 unresolved discussions (waiting on @hdkshingala)

___
*[`manifests/charts/aperture-operator/values.yaml` line 350 at r1](https://reviewable.io/reviews/fluxninja/aperture/206#-NAdJqAR1LnrjDOisjiR:-NAdJqASDUFf1k-baEkX:b4b7wbm) ([raw file](https://github.com/fluxninja/aperture/blob/225f689646368bc43431d8025b219b7858d1b28e/manifests/charts/aperture-operator/values.yaml#L350)):*
> ```YAML
>       insecureSkipVerify: ~
>       caFile: ~
>     apiKeySecret:
> ```

Controllers and Agents use different API keys, why this is unified?
___
*[`manifests/charts/aperture-operator/values.yaml` line 391 at r1](https://reviewable.io/reviews/fluxninja/aperture/206#-NAdKLhxD_PcADT6pb8l:-NAdKLhxD_PcADT6pb8m:b-kgu5m0) ([raw file](https://github.com/fluxninja/aperture/blob/225f689646368bc43431d8025b219b7858d1b28e/manifests/charts/aperture-operator/values.yaml#L391)):*
> ```YAML
>   ## @param agent.batchPrerollup.sendBatchSize Size of a batch in prerollup processor which after hit, will trigger it to be sent.
>   ##
>   batchPrerollup:
> ```

I am a little bit confused, does this PR include changes from https://github.com/fluxninja/aperture/pull/159 as well?

**Review by hdkshingala — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/fluxninja/aperture/206)* status: all files reviewed, 2 unresolved discussions (waiting on @kwapik)

___
*[`manifests/charts/aperture-operator/values.yaml` line 350 at r1](https://reviewable.io/reviews/fluxninja/aperture/206#-NAdJqAR1LnrjDOisjiR:-NAdL53d65qqwH1-sUYH:bo566fx) ([raw file](https://github.com/fluxninja/aperture/blob/225f689646368bc43431d8025b219b7858d1b28e/manifests/charts/aperture-operator/values.yaml#L350)):*
<details><summary><i>Previously, kwapik (Krzysztof Kwapisiewicz) wrote…</i></summary><blockquote>

Controllers and Agents use different API keys, why this is unified?
</blockquote></details>

Now as we have 2 different Custom Resources for the Agent and Controller, this block is only for Agent CR. We have a similar block for Controller below this.
___
*[`manifests/charts/aperture-operator/values.yaml` line 391 at r1](https://reviewable.io/reviews/fluxninja/aperture/206#-NAdKLhxD_PcADT6pb8l:-NAdLRzR7EHuv6MgzDDH:b-jjvuxw) ([raw file](https://github.com/fluxninja/aperture/blob/225f689646368bc43431d8025b219b7858d1b28e/manifests/charts/aperture-operator/values.yaml#L391)):*
<details><summary><i>Previously, kwapik (Krzysztof Kwapisiewicz) wrote…</i></summary><blockquote>

I am a little bit confused, does this PR include changes from https://github.com/fluxninja/aperture/pull/159 as well?
</blockquote></details>

Oh I missed that Jakub synced the operator repo changes in here as it wasn't merged yet.
But these changes will stay as the values file structure is changed.
I'll check Jakub's changes and drop commit for this.

**Review by kwapik — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/fluxninja/aperture/206)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @hdkshingala)

___
*[`manifests/charts/aperture-operator/values.yaml` line 391 at r1](https://reviewable.io/reviews/fluxninja/aperture/206#-NAdKLhxD_PcADT6pb8l:-NAdQZeY3VqZqkEyK3Wi:buol72n) ([raw file](https://github.com/fluxninja/aperture/blob/225f689646368bc43431d8025b219b7858d1b28e/manifests/charts/aperture-operator/values.yaml#L391)):*
<details><summary><i>Previously, hdkshingala wrote…</i></summary><blockquote>

Oh I missed that Jakub synced the operator repo changes in here as it wasn't merged yet.
But these changes will stay as the values file structure is changed.
I'll check Jakub's changes and drop commit for this.
</blockquote></details>

To sum up discussion outside GH - we're merging this PR as is and then rebasing and fixing #159 on top of that.
